src/python/gudhi/representations/vector_methods.py

```diff
@@ -574,3 +575,140 @@ def __call__(self, diag):
+    def __init__(self, quantiser, weighting_method="cloud", contrast="gaussian"):
+        """
+        Constructor for the Atol measure vectorisation class.
+
+        Parameters:
+            quantiser (Object): Object with `fit` (sklearn API consistent) and `cluster_centers` and `n_clusters`
+                attributes, e.g. sklearn.cluster.KMeans. It will be fitted when the Atol object function `fit` is called.
+            weighting_method (string): constant generic function for weighting the measure points
+                choose from {"cloud", "iidproba"}
+                (default: constant function, i.e. the measure is seen as a point cloud by default).
+                This will have no impact if weights are provided along with measures all the way: `fit` and `transform`.
+            contrast (string): constant function for evaluating proximity of a measure with respect to centers
+                choose from {"gaussian", "laplacian", "indicator"}
+                (default: gaussian contrast function, see page 3 in the ATOL paper).
+        """
+        self.quantiser = quantiser
+        self.contrast = {
+            "gaussian": _gaus_contrast,
+            "laplacian": _lapl_contrast,
+            "indicator": _indicator_contrast,
+        }.get(contrast, _gaus_contrast)
+        self.weighting_method = {
+            "cloud"   : _cloud_weighting,
+            "iidproba": _iidproba_weighting,
+        }.get(weighting_method, _cloud_weighting)
+
+    def fit(self, X, y=None, sample_weight=None):
+        """
+        Calibration step: fit centers to the sample measures and derive inertias between centers.
+
+        Parameters:
+            X (list N x d numpy arrays): input measures in R^d from which to learn center locations and inertias
+                (measures can have different N).
+            y: Ignored, present for API consistency by convention.
+            sample_weight (list of numpy arrays): weights for each measure point in X, optional.
+                If None, the object's weighting_method will be used.
+
+        Returns:
+            self
+        """
+        if not hasattr(self.quantiser, 'fit'):
+            raise TypeError("quantiser %s has no `fit` attribute." % (self.quantiser))
+        if sample_weight is None:
+            sample_weight = np.concatenate([self.weighting_method(measure) for measure in X])
+
+        measures_concat = np.concatenate(X)
+        self.quantiser.fit(X=measures_concat, sample_weight=sample_weight)
+        self.centers = self.quantiser.cluster_centers_
+        if self.quantiser.n_clusters == 1:
+            dist_centers = pairwise.pairwise_distances(measures_concat)
+            np.fill_diagonal(dist_centers, 0)
+            self.inertias = np.max(dist_centers)/2
+        else:
+            dist_centers = pairwise.pairwise_distances(self.centers)
+            np.fill_diagonal(dist_centers, np.inf)
+            self.inertias = np.min(dist_centers, axis=0)/2
+        return self
```

Comment:
It happens that two quantiser centers are identical, which leads to 0s in self.inertias and that is a problem I could address here...?

Reply:
That sounds like a good idea. Are they exactly identical, or can they be very close but slightly apart? Can there be an arbitrary number of identical centers? Do you already know how you want to handle this problem?

Reply:
It happens that they can be very close but slightly apart, or that they are exactly identical. And there can be an arbitrary number of identical centers as far as I can tell. I am unsure as to how to handle this.

Options that I think of include:
1- removing centers that are identical - it actually sounds rather solid but we lose control of the number of centers and does not feel _natural_.
2- recasting the resulting inertias that are too close to zero - which is what prompts error when computing the contrast functions over measures. I actually like the simplicity of this. In a way it says that the quantised mean measure is supported on points that can have multiplicity higher than 1.
3- ensuring ahead of quantising that the calibrating measure points are sufficiently distinct (using some sort of tree-trimming), so that the space quantising will pick sufficiently different cluster centers; even though this could work well in practice, it is rather shaky as it assumes the quantiser will not mess up and adds an extra and arbitrary trimming step that could obfuscate the signal in the end.

Reply:
I don't understand what 2- means, but anyway, I am also ok with merging the current version and improving it once you have settled on a way to handle it, if the problem doesn't occur too often.

Reply:
Yes it is rare and should be even rarer now, I'm ok with merging too.

Reply:
Let's wait and see if Vincent has any comments, he can handle the merging then.